src/core/quantum_circuit.jl

```diff
@@ -605,6 +653,7 @@ end
 function Base.show(io::IO, circuit::QuantumCircuit, padding_width::Integer = 10)
     println(io, "Quantum Circuit Object:")
     println(io, "   qubit_count: $(get_num_qubits(circuit)) ")
+    println(io, "   bit_count: $(get_num_bits(circuit)) ")
```

Comment:
We already have an overloaded equality: `Base.isequal(QuantumCircuit, QuantumCircuit)` (which needs to include bit_count, now that I think of it)

Otherwise, printing bit_count is a matter of taste IMO. @pleroux0 ?

Reply:
Since bit count needs to be equal for circuits to be equal, I would say that printing it is relevant.

Reply:
Printing it makes sense... TIL about [the fix flag](https://documenter.juliadocs.org/stable/man/doctests/#Fixing-Outdated-Doctests) for Julia doctests so hopefully we can use that to cut down on the toil.